src/OrleansRuntime/Messaging/MessageCenter.cs

```diff
@@ -37,9 +37,13 @@ public bool TryDeliverToProxy(Message msg)
 
         public IMessagingConfiguration MessagingConfiguration { get; private set; }
 
-        public MessageCenter(IPEndPoint here, int generation, IMessagingConfiguration config, ISiloPerformanceMetrics metrics = null)
+        public MessageCenter(SiloInitializationParameters silo, NodeConfiguration nodeConfig, IMessagingConfiguration config, ISiloPerformanceMetrics metrics = null)
         {
-            Initialize(here, generation, config, metrics);
+            this.Initialize(silo.SiloAddress.Endpoint, nodeConfig.Generation, config, metrics);
+            if (nodeConfig.IsGatewayNode)
+            {
+                this.InstallGateway(nodeConfig.ProxyGatewayEndpoint);
```

Comment:
Good feedback, there are lots of places where I've been trying to cut the circular deps. I think we should address them in a later PR if possible. In this specific case, it does not look as though there will be an issue.

I think the code would be clearer if I were to inline both Initialize & InstallGateway into the constructor - how about you?